dali/operators/reader/loader/tar_utils.cc

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/reader/loader/tar_utils.h"
+#include <libtar.h>
+#include <algorithm>
+#include <cstdarg>
+#include <cstdlib>
+#include <string>
+#include <tuple>
+#include <utility>
+#include "dali/core/error_handling.h"
+
+namespace dali {
+namespace detail {
```

Comment:
If these are not used outside of this translation unit, then use an anonymous namespace - it will guarantee that the symbols won't leak to other translation units.

Reply:
No, I mean the link symbols. `namespace {` has special semantics, making everything inside private to the translation units. I raise this concern bacause use global variables and highly collision-prone names like `instances` or `instances_mutex`. If you don't place them in an anonymous namespace (or specify internal linkage with `static` keyword), then these names will be visible in the whole library (dali/operators) and there would be conflicts, should someone use the same names elsewhere.